### PR TITLE
Resolves #2 Adapter for storing downloaded diagrams into Antora content catalog

### LINKED ISSUE #2: Properly store generated PlantUML images directly in Antora content folder instead of output directory.
This is a placeholder for [antora#189](https://gitlab.com/antora/antora/issues/189)

### PR DESCRIPTION
**Pull request by eshepelyuk**

### CODE REVIEW

File: src/asciidoctor-plantuml.js [hunk 1]
```diff
@@ -86,7 +86,30 @@ function plantumlBlock (context) {
...
+        basename: image.basename,
+        relative: image.basename
+      }
+    })
+  }
+})
+
 module.exports.register = function register (registry, context = {}) {
+  // patch context in case of Antora
+  if (typeof context.contentCatalog !== 'undefined' && typeof context.file !== 'undefined') {
```
Comment (Mogztter) [inline src/asciidoctor-plantuml.js]: I think we should add a test with a fake Antora context to make sure that this test is valid.
Also maybe we should check that `context.contentCatalog.addFile` is a function.

@mojavelinux Is there a more accurate way to detect an Antora context ?

File: src/asciidoctor-plantuml.js [hunk 2]
```diff
@@ -86,7 +86,30 @@ function plantumlBlock (context) {
   }
 }
 
+const antoraAdapter = (file, contentCatalog) => ({
```
Comment (Mogztter) [inline src/asciidoctor-plantuml.js]: As mentioned in https://github.com/eshepelyuk/asciidoctor-plantuml-antora/pull/1#issuecomment-450377212 I don't think we should add this code in this extension but that's just my opinion.
@mojavelinux What do you think ?

File: src/asciidoctor-plantuml.js (hunk 1 above)
Comment (eshepelyuk) [reply src/asciidoctor-plantuml.js]: I agree with test, it's in my _backlog_ as soon as I some more spare time :)

**Comment by mojavelinux** [comment]
Btw, once we get into the specification for AsciiDoc, I'm hoping that this will no longer be just an Antora thing, but rather the VFS for the AsciiDoc extension API. So we really shouldn't think of this as being specific to Antora, but rather an exploration for what could become a standard. And for it to become a standard, it has to first be proved.

**Comment by mojavelinux** [comment]
(After all, we have to have some way for extensions to be able to contribute files back into the Asciidoctor context so that Asciidoctor can know about them and make references to them).

**Pull request merged**